sign/sign.go

```diff
@@ -397,6 +404,137 @@ func (s *Signer) IsImageSigned(imageRef string) (bool, error) {
 	return len(signatures) > 0, nil
 }
 
+const maxWorkers = 100
+
+// ImagesSigned verifies if the provided image references are signed. It
+// returns a sync map where the key is the ref and the value is a boolean which
+// indicates if the image is signed or not. The method runs highly parallel.
+func (s *Signer) ImagesSigned(ctx context.Context, refs ...string) (*sync.Map, error) {
+	s.log().Debug("Parsing references")
+	repos := []name.Repository{}
+	for _, ref := range refs {
+		parsedRef, err := s.impl.ParseReference(ref)
+		if err != nil {
+			return nil, fmt.Errorf("parsing image reference: %w", err)
+		}
+
+		repos = append(repos, parsedRef.Context())
+	}
+
+	s.log().Debug("Building transports")
+	transports, count, err := s.transportsForRefs(ctx, repos...)
+	if err != nil {
+		return nil, fmt.Errorf("build transports: %w", err)
+	}
+	s.log().Debugf("Built %d transports for %d refs", count, len(refs))
+
+	s.log().Debugf("Checking if refs are signed")
+	t := throttler.New(maxWorkers, len(refs))
+	res := &sync.Map{}
+	for i, repo := range repos {
+		go func(repo name.Repository, i int) {
+			trans, ok := transports.Load(repo)
+			if !ok {
+				t.Done(fmt.Errorf("no transport found for repo: %s", repo.String()))
+				return
+			}
+			tr, ok := trans.(http.RoundTripper)
+			if !ok {
+				t.Done(fmt.Errorf("transport has wrong type: %v", tr))
+				return
+			}
+
+			digest, err := s.impl.Digest(refs[i], crane.WithTransport(tr))
+			if err != nil {
+				t.Done(fmt.Errorf("get digest for image reference: %w", err))
+				return
+			}
+
+			if _, err := s.impl.Digest(repoDigestToSig(repo, digest), crane.WithTransport(tr)); err != nil {
+				if transportErr, ok := err.(*transport.Error); ok && len(transportErr.Errors) > 0 {
+					if transportErr.Errors[0].Code == transport.ManifestUnknownErrorCode {
+						res.Store(refs[i], false)
+						t.Done(nil)
```

Comment:
Why we don't have any error message here?

Reply:
Because the error (`ManifestUnknownErrorCode`) indicates that the signature does not exist remotely, which is a valid case.